tests/pausing_test.go

```diff
@@ -53,136 +52,98 @@ import (
+					HTTPGet: &k8sv1.HTTPGetAction{
+						Path: "/something",
+						Port: intstr.FromInt(8080),
+					},
+				},
+				InitialDelaySeconds: 120,
+				TimeoutSeconds:      120,
+				PeriodSeconds:       120,
+				SuccessThreshold:    1,
+				FailureThreshold:    1,
+			}
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 90)
+
+			By("Pausing it")
+			err := virtClient.VirtualMachineInstance(vmi.Namespace).Pause(context.Background(), vmi.Name, &v1.PauseOptions{})
+			Expect(err).To(MatchError(ContainSubstring("Pausing VMIs with LivenessProbe is currently not supported")))
 		})
 
-		When("paused via virtctl with --dry-run flag", func() {
-			It("[test_id:7671]should not paused", func() {
-
-				command := clientcmd.NewRepeatableVirtctlCommand("pause", "vmi", "--dry-run", "--namespace", vmi.Namespace, vmi.Name)
-				Expect(command()).To(Succeed())
-				By("Checking that VMI remains running")
-				Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstancePaused))
-			})
+		It("[test_id:7671]should not pause with dry run", func() {
+			err := virtClient.VirtualMachineInstance(vmi.Namespace).Pause(context.Background(), vmi.Name, &v1.PauseOptions{DryRun: []string{metav1.DryRunAll}})
+			Expect(err).ToNot(HaveOccurred())
+			By("Checking that VMI remains running")
+			Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstancePaused))
 		})
 
-		When("unpaused via virtctl with --dry-run flag", func() {
-			It("[test_id:7672]should not unpaused", func() {
-
-				command := clientcmd.NewRepeatableVirtctlCommand("pause", "vmi", "--namespace", vmi.Namespace, vmi.Name)
-				Expect(command()).To(Succeed())
-				Eventually(matcher.ThisVMI(vmi), 30*time.Second, time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
+		It("[test_id:7672]should not unpause with dry run", func() {
+			err := virtClient.VirtualMachineInstance(vmi.Namespace).Pause(context.Background(), vmi.Name, &v1.PauseOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(matcher.ThisVMI(vmi), 30*time.Second, time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
 
-				command = clientcmd.NewRepeatableVirtctlCommand("unpause", "vmi", "--dry-run", "--namespace", vmi.Namespace, vmi.Name)
-				Expect(command()).To(Succeed())
+			err = virtClient.VirtualMachineInstance(vmi.Namespace).Unpause(context.Background(), vmi.Name, &v1.UnpauseOptions{DryRun: []string{metav1.DryRunAll}})
+			Expect(err).ToNot(HaveOccurred())
 
-				By("Checking that VMI remains paused")
-				Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
-			})
+			By("Checking that VMI remains paused")
+			Consistently(matcher.ThisVMI(vmi), 5*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstancePaused))
 		})
 
 		It("should not appear as ready when paused", func() {
```

Comment:
How about adding this expectation to another test?
Starting a VM and pausing it once for every condition seems awfully expensive.

Reply:
Good suggestion, addressed!